### DISCUSSION TITLE
Add ability to use custom Certificate Authority Pool for Cloudflare Tunnels

### DISCUSSION THREAD

**Discussion by donicrosby** [homeassistant-apps/app-cloudflared#669] [Ideas]
I have a local private CA (step-ca) that I use to author TLS certificates in my network

I would like to have Home Assistant use HTTPS for all network connections using the certificate made from my certificate authority.

Cloudflared works fine with HTTP but I would like to be able to add a custom CA Root Certificate to the container so that I am able to tell Cloudflare to check that the certificate is signed by my private CA's Root. 

Cloudflare already has an option to do this with the Certificate Authority Pool option in their UI. I just need a way to inject the cert into the container and tell Cloudflared to use it to validate the certificate against.

A optional configuration option where you put the certificate data and it's mapped to a static file location in the container would be perfect for this.

**Comment by KNGP14** [comment]
corresponding cloudflare docs
[https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/configure-tunnels/cloudflared-parameters/origin-parameters/#capool](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/configure-tunnels/cloudflared-parameters/origin-parameters/#capool)

**Comment by elcajon** [comment]
I guess this is a pretty rare use case, to keep the add-on as simple as possible for most "regular" users, but you could achieve this by using the remote managed tunnel. Simply configure the CA pool using the Cloudflare Zero Trust Dashboard and pass the tunnel token to our add-on.
